opencensus/stats/exporters/prometheus_exporter.py

```diff
@@ -183,26 +184,30 @@ def to_metric(self, desc, tag_values, agg_data):
                         aggregation_data_module.DistributionAggregationData):
 
             assert(agg_data.bounds == sorted(agg_data.bounds))
-            points = {}
-            cum_count = 0
+            # buckets are a list of bucket. Each bucket is another list with
+            # a pair of bucket name and value, or a triple of bucket name,
+            # value, and exemplar. buckets need to be in order.
+            buckets = []
+            cum_count = 0  # Prometheus buckets expect cumulative count.
             for ii, bound in enumerate(agg_data.bounds):
                 cum_count += agg_data.counts_per_bucket[ii]
-                points[str(bound)] = cum_count
+                bucket = [str(bound), cum_count]
+                buckets.append(bucket)
             # Prometheus requires buckets to be sorted, and +Inf present.
             # In OpenCensus we don't have +Inf in the bucket bonds so need to
             # append it here.
-            points["+Inf"] = agg_data.count_data
+            buckets.append(["+Inf", agg_data.count_data])
             metric = HistogramMetricFamily(name=metric_name,
                                            documentation=metric_description,
                                            labels=label_keys)
             metric.add_metric(labels=tag_values,
-                              buckets=list(points.items()),
+                              buckets=buckets,
                               sum_value=agg_data.sum,)
             return metric
 
         elif isinstance(agg_data,
                         aggregation_data_module.SumAggregationDataFloat):
-            metric = UntypedMetricFamily(name=metric_name,
+            metric = UnknownMetricFamily(name=metric_name,
```

Comment:
What is the `SummaryMetricFamily` then?

Reply:
`SummaryMetricFamily` is for `Summary` metrics. `Sum` metrics should be converted to `UnknownMetricFamily`.